### PR TITLE
Load Unload lenght from Marlin config

### PR DESCRIPTION
**Pull request by Hagridpon3**
Adding values ​​from Configuration_A3ides_2209_MINI_adv to menu_vars.h for simple settings

### CODE REVIEW

File: src/gui/menu_vars.cpp [hunk 1]
```diff
@@ -102,3 +103,11 @@ constexpr const char Z_home_gcode[] = {
...
 
 }
+
+constexpr const int32_t filament_change_slow_load_lenght = FILAMENT_CHANGE_SLOW_LOAD_LENGTH;
+constexpr const int32_t filament_change_fast_load_lenght = FILAMENT_CHANGE_FAST_LOAD_LENGTH;
+constexpr const int32_t filament_change_slow_purge_lenght = 40;
+constexpr const float filament_unload_mini_lengt = 392.0F;
+
+constexpr const int32_t filament_change_full_load_lenght = filament_change_fast_load_lenght + filament_change_slow_load_lenght;
+constexpr const int32_t filament_change_full_purge_load_lenght = filament_change_full_load_lenght + filament_change_slow_purge_lenght;
```
Comment (mkbel) [inline src/gui/menu_vars.cpp]: Fix spelling lenght -> length

File: src/gui/menu_vars.h [hunk 2]
```diff
@@ -50,6 +50,13 @@ extern const char X_home_gcode[];
 extern const char Y_home_gcode[];
 extern const char Z_home_gcode[];
 
+extern const int32_t filament_change_slow_load_lenght;
+extern const int32_t filament_change_fast_load_lenght;
+extern const int32_t filament_change_full_load_lenght;
+extern const int32_t filament_change_slow_purge_lenght;
+extern const int32_t filament_change_full_purge_load_lenght;
```
Comment (mkbel) [inline src/gui/menu_vars.h]: Fix spelling lenght -> length

File: src/gui/menu_vars.cpp (hunk 1 above)
Comment (Hagridpon3) [reply src/gui/menu_vars.cpp]: ok

File: src/gui/menu_vars.h [hunk 3]
```diff
@@ -50,6 +50,13 @@ extern const char X_home_gcode[];
...
+extern const float filament_unload_mini_lengt;
```
Comment (mkbel) [inline src/gui/menu_vars.h]: Fix spelling lengt -> length

File: src/gui/menu_vars.h (hunk 2 above)
Comment (Hagridpon3) [reply src/gui/menu_vars.h]: ok

File: src/gui/menu_vars.h (hunk 3 above)
Comment (Hagridpon3) [reply src/gui/menu_vars.h]: ok

File: src/gui/dialogs/window_dlg_change.c [hunk 4]
```diff
@@ -81,9 +81,9 @@ static int f_CH_WAIT_E_POS__RAMMING(_dlg_vars *p_vars, _dlg_ld_vars *additional_
 static int f_CH_WAIT_E_POS__UNLOADING(_dlg_vars *p_vars, _dlg_ld_vars *additional_vars) {
     // 8 ... -392
     float pos = additional_vars->p_marlin_vars->pos[MARLIN_VAR_INDEX_E] - additional_vars->e_start;
-    if (pos < -391.9F)
+    if (pos < (filament_unload_mini_lengt - 1.0F))
```
Comment (mkbel) [inline src/gui/dialogs/window_dlg_change.c]: Fix spelling lengt -> length

File: src/gui/dialogs/window_dlg_change.c [hunk 5]
```diff
@@ -126,26 +126,26 @@ static int f_CH_WAIT_E_POS__INSERTING(_dlg_vars *p_vars, _dlg_ld_vars *additiona
...
 }
 
 static int f_CH_WAIT_E_POS__PURGING(_dlg_vars *p_vars, _dlg_ld_vars *additional_vars) {
     //wait E pos >= 400
     float pos = additional_vars->p_marlin_vars->pos[MARLIN_VAR_INDEX_E] - additional_vars->e_start;
-    if ((pos >= 400) && (marlin_motion() == 0))
+    if ((pos >= filament_change_full_purge_load_lenght) && (marlin_motion() == 0))
         p_vars->phase++;
-    return 100 * (pos - 360) / (400 - 360);
+    return 100 * (pos - filament_change_full_load_lenght) / (filament_change_full_purge_load_lenght - filament_change_full_load_lenght);
```
Comment (mkbel) [inline src/gui/dialogs/window_dlg_change.c]: Fix spelling lenght -> length

**Pull request merged**